src/junit.rs

```diff
@@ -10,36 +10,42 @@ use quick_xml::reader::Reader;
 use crate::testrun::{Outcome, Testrun};
 
 // from https://gist.github.com/scott-codecov/311c174ecc7de87f7d7c50371c6ef927#file-cobertura-rs-L18-L31
-fn attributes_map(attributes: Attributes) -> HashMap<String, String> {
-    return attributes
-        .filter_map(|a| {
-            if let Ok(attr) = a {
-                let bytes = attr.value.into_owned();
-                let value = String::from_utf8(bytes).unwrap();
-                let key = String::from_utf8(attr.key.local_name().as_ref().to_vec()).unwrap();
-                Some((key, value))
-            } else {
-                None
-            }
-        })
-        .collect::<HashMap<_, _>>();
+fn attributes_map(attributes: Attributes) -> Result<HashMap<String, String>, pyo3::PyErr> {
+    let mut attr_map = HashMap::new();
+    for attribute in attributes {
+        if let Ok(attr) = attribute {
+            let bytes = attr.value.into_owned();
+            let value = String::from_utf8(bytes)?;
```

Comment:
(I'm writing this after writing that one above ^) so the ?; is a less verbose way of handling errors, so if I'm reading this right, this would unpack if `Ok`, else it'll return the error that `String::from_utf8` provides right? 

If so, how would that trickle down to PyErr?

Reply:
`?` is different from unwrap because unwrap will `panic` but `?` will not panic it will propagate the error up to the caller of the function, so it's more appropriate than unwrap for our use case